tests/nopytest_convertReferenceCatalog.py

```diff
@@ -55,77 +55,73 @@ class TestConvertReferenceCatalogParallel(convertReferenceCatalogTestBase.Conver
+                                                                   withFullPositionInformation=True)
+        # use a very small HTM pixelization depth to ensure there will be collisions when
+        # ingesting the files in parallel
+        depth = 2
+        config.dataset_config.indexer.active.depth = depth
+        # np.savetxt prepends '# ' to the header lines, so use a reader that understands that
+        config.file_reader.format = 'ascii.commented_header'
+        config.n_processes = 2  # use multiple cores for this test only
+        config.id_name = 'id'  # Use the ids from the generated catalogs
+        repoPath = os.path.join(self.outPath, "output_multifile_parallel")
+
+        # Convert the input data files to our HTM indexed format.
+        dataTempDir = tempfile.TemporaryDirectory()
+        dataPath = dataTempDir.name
+        converter = ConvertReferenceCatalogTask(output_dir=dataPath, config=config)
+        converter.run([skyCatalogFile1, skyCatalogFile2])
+
+        # Make a temporary butler to ingest them into.
+        butler = self.makeTemporaryRepo(repoPath, config.dataset_config.indexer.active.depth)
+        dimensions = [f"htm{depth}"]
+        datasetType = DatasetType(config.dataset_config.ref_dataset_name,
+                                  dimensions,
+                                  "SimpleCatalog",
+                                  universe=butler.registry.dimensions,
+                                  isCalibration=False)
+        butler.registry.registerDatasetType(datasetType)
+
+        # Ingest the files into the new butler.
+        run = "testingRun"
+        htmTableFile = os.path.join(dataPath, "filename_to_htm.ecsv")
+        ingest_files(repoPath,
+                     config.dataset_config.ref_dataset_name,
+                     run,
+                     htmTableFile,
+                     transfer="auto")
+
+        # Test if we can get back the catalogs, with a new butler.
+        butler = lsst.daf.butler.Butler(repoPath)
+        datasetRefs = list(butler.registry.queryDatasets(config.dataset_config.ref_dataset_name,
+                                                         collections=[run]).expanded())
+        handlers = []
+        for dataRef in datasetRefs:
+            handlers.append(DeferredDatasetHandle(butler=butler, ref=dataRef, parameters=None))
+        loaderConfig = ReferenceObjectLoader.ConfigClass()
+        loader = ReferenceObjectLoader([dataRef.dataId for dataRef in datasetRefs],
+                                       handlers,
+                                       name="testRefCat",
+                                       config=loaderConfig,
+                                       log=self.logger)
+        self.checkAllRowsInRefcat(loader, skyCatalog1, config)
+        self.checkAllRowsInRefcat(loader, skyCatalog2, config)
+
+        inTempDir1.cleanup()
+        inTempDir2.cleanup()
+        dataTempDir.cleanup()
```

Comment:
Path cleanups need to happen in `tearDown`

Reply:
This is what was already there, and these aren't class variables.

Reply:
Then we should probably make them so, otherwise if the test fails, it won't cleanup.

Reply:
An alternative that should make @parejkoj happy while keeping them isolated to the test that actually uses them is to use the original `TemporaryDirectory` calls as (triply-nested 😬) context managers.

Reply:
You can put all the context managers inside the same `with`, so you don't have to nest them. That would work.

Reply:
@parejkoj , could you give me a code snippet for that?

Reply:
`with tempfile.TemporaryDirectory() as path1`, with tempfile.TemporaryDirectory() as path2:`, etc.

Reply:
This doesn't work. It looks like you aren't supposed to repeat `with`, but even after fixing that, I'm not getting a `TemporaryDirectory()` object with this--I just get a string with the directory name. I'm just going to go back to the option of making these class variables and adding `setup` and `teardown` functions.

Reply:
Sorry, yes, you don't repeat `with`. When used as a context manager, the directory name is the target of the `with`.

See the docs:

https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory